AppManager/app_manager_server.py

```diff
@@ -658,6 +675,168 @@ def create_java_start_cmd(app_name, port, load_balancer_host, max_heap,
   return ' '.join(cmd)
 
 
+def get_instance_nodes(instance_key):
+  """ Fetches the ZooKeeper nodes for a given instance key.
+
+  Args:
+    instance_key: A string specifying the revision key and port of an instance.
+  Returns:
+    A tuple of strings specifying the instances nodes.
+  """
+  # Instances are indexed both by version and by machine.
+  instance_node_by_machine = '/'.join([machine_instances_node, instance_key])
+
+  revision_key, port = instance_key.split(':')
+  revision_parts = revision_key.split(VERSION_PATH_SEPARATOR)
+  version_key = VERSION_PATH_SEPARATOR.join(revision_parts[:3])
+  instance_node_by_version = '/appscale/instances_by_version/{}/{}'.format(
```

Comment:
should this path be a constant?

Reply:
@obino In the design I'm aiming at, we'll need two nodes to keep track of instance assignments (by machine and by version) and two nodes to keep track of instance registrations (by machine and by version).

The AppManager will need a watch on assignments_by_machine. The router will need a watch on instances_by_version. The autoscaler will need a watch on assignments_by_version (to determine if configured constraints are being met) and on instances_by_machine (to determine if assignments have been fulfilled).

Reply:
@cdonati got it. Would you mind if we get a quick presentation of this design after one of our daily? I think it's interesting to share :)